internal/controller/argocdcommitstatus_controller.go

```diff
@@ -324,6 +332,17 @@ func (r *ArgoCDCommitStatusReconciler) updateAggregatedCommitStatus(ctx context.
 	gvk := promoterv1alpha1.GroupVersion.WithKind(kind)
 	controllerRef := metav1.NewControllerRef(&argoCDCommitStatus, gvk)
 
+	sw := &strings.Builder{}
+	t, err := template.New("url").Parse(argoCDCommitStatus.Spec.URLTemplate)
+	if err != nil {
+		return fmt.Errorf("failed to parse URL template: %w", err)
+	}
+	err = t.Execute(sw, map[string]any{"apps": apps})
```

Comment:
TODO: add `targetBranch` var, since that'll be the same for all apps in the list.